dipy/denoise/gibbs.py

```diff
@@ -0,0 +1,310 @@
+        PTV = PTV + np.absolute(xs[:, (nn+1+n):(-nn-1+n)] -
+                                xs[:, (nn+2+n):(-nn+n)])
+        NTV = NTV + np.absolute(xs[:, (nn+1-n):(-nn-1-n)] -
+                                xs[:, (nn-n):(-nn-2-n)])
+
+    if a:
+        return PTV, NTV
+    else:
+        return PTV.T, NTV.T
+
+
+def gibbs_removal_1d(x, a=0, nn=3):
+    """ Decreases gibbs ringing along an axis 'a' using fourier sub-shifts.
+
+    Parameters
+    ----------
+    x : 2D ndarray
+        Matrix x.
+    a : int (0 or 1)
+        Axis along which gibbs oscilations will be reduced. Default a is set
+        to 0 (i.e. gibbs are reduce along axis y).
+    nn : int, optional
+        Number of neighbour points to access local TV (see note). Default is
+        set to 3.
+
+    Returns
+    -------
+    xc : 2D ndarray
+        Matrix with gibbs oscilantions reduced along an axis 'a'.
+
+    Note
+    ----
+    This function decreases the effects of gibbs oscilations based on the
+    analysis of local total variation (TV). Although artefact correction is
+    done based on two adjanced points for each voxel, total variation should be
+    accessed in a larger range of neigbors. If you want to adjust the number
+    of the neigbors to be considered in TV calculation please change
+    parameter nn.
+    """
+    ssamp = np.linspace(0.02, 0.9, num=45)
+
+    if a:
+        xs = x.copy()
+    else:
+        xs = x.T.copy()
+
+    # TV for shift zero (baseline)
+    TVR, TVL = image_tv(xs, a=1, nn=nn)
+    TVP = np.minimum(TVR, TVL)
+    TVN = TVP.copy()
+
+    # Find optimal shift for gibbs removal
+    ISP = xs.copy()
+    ISN = xs.copy()
+    SP = np.zeros(xs.shape)
+    SN = np.zeros(xs.shape)
+    N = xs.shape[1]
+    c = np.fft.fftshift(np.fft.fft2(xs))
+    k = np.linspace(-N/2, N/2-1, num=N)
+    k = (2.0j * np.pi * k) / N
+    for s in ssamp:
+        # Access positive shift for given s
+        Img_p = abs(np.fft.ifft2(np.fft.fftshift(c * np.exp(k*s))))
+        TVSR, TVSL = image_tv(Img_p, a=1, nn=nn)
+        TVS_p = np.minimum(TVSR, TVSL)
+
+        # Access negative shift for given s
+        Img_n = abs(np.fft.ifft2(np.fft.fftshift(c * np.exp(-k*s))))
+        TVSR, TVSL = image_tv(Img_n, a=1, nn=nn)
+        TVS_n = np.minimum(TVSR, TVSL)
+
+        # Update positive shift params
+        ISP[TVP > TVS_p] = Img_p[TVP > TVS_p]
+        SP[TVP > TVS_p] = s
+        TVP[TVP > TVS_p] = TVS_p[TVP > TVS_p]
+
+        # Update negative shift params
+        ISN[TVN > TVS_n] = Img_n[TVN > TVS_n]
+        SN[TVN > TVS_n] = s
+        TVN[TVN > TVS_n] = TVS_n[TVN > TVS_n]
+
+    # check non-zero sub-voxel shifts
+    idx = np.nonzero(SP + SN)
+
+    # use positive and negative optimal sub-voxel shifts to interpolate to
+    # original grid points
+    xs[idx] = (ISP[idx] - ISN[idx])/(SP[idx] + SN[idx])*SN[idx] + ISN[idx]
+
+    if a:
+        return xs
+    else:
+        return xs.T
+
+
+def gibbs_weigthing_functions(shape):
```

Comment:
How about renaming this `gibbs_weighting`? Or even just `weighting`? It is already in the `gibbs` module.